File: Nov2020/19Nov/ClimbingStairs.java
/*
Problem Name: 70. Climbing Stairs
Problem Link: https://leetcode.com/problems/climbing-stairs/
*/
class Solution {
    public int climbStairs(int n) {
        if(n == 1)
            return 1;
        if(n == 2)
            return 2;
        int step0 = 1, step1 = 2, step2 = 0;
        for(int i=2;i<n;i++){
            step2 = step0 + step1;
            step0 = step1;
            step1 = step2;
        }
        return step2;
    }
}

File: Nov2020/19Nov/ClimbingStairs.cpp
/*
Problem name: Climbing Stairs
Problem link: https://leetcode.com/problems/climbing-stairs/
*/

class Solution {
public:
    int climbStairs(int n) {
        int op[n+1];
        op[0]=1;
        op[1]=1;
        for(int i=2;i<=n;++i){
            op[i]=op[i-1]+op[i-2];
        }
        return op[n];
    }
};
